src/javascript/components/CreateAgreementForm.jsx

```diff
@@ -1,36 +1,77 @@
 // @flow
-import React from 'react';
+import React, { Component } from 'react';
 
-const CreateAgreementForm = (props: {onSubmit: Function, error: string}) => {
-  const { onSubmit, error } = props;
-
-  let buyer,
-      seller,
-      price;
+type Props = {
+  agreementPending: string,
+  error: string,
+}
 
-  return (
-    <div>
-      <h1>Trustless Escrow</h1>
+export default class CreateAgreementForm extends Component {
+  state: {
+    buyer: string;
+    seller: string;
+    price: string;
+  }
 
-      <form onSubmit={e => {
-        e.preventDefault();
-        if (!buyer.value.trim() || !seller.value.trim()) {
-          return;
-        };
+  constructor(props: Props) {
+    super(props);
+    // going to start by giving input state to the component
+    // until the value is submitted, then giving the values to the
+    // global store
+    this.state = {
+      buyer: '',
+      seller: '',
+      price: '',
+    }
+    // flow error with binding class methods in constructor
+    // can be removed if we use ES7 stage 2 arrow functions to
+    // encapsulate scope
+    // there are other methods, but they reduce performance
+    const self: Object = this;
 
-        onSubmit(buyer.value, seller.value, price.value);
-      }}>
+    self.handleChange = this.handleChange.bind(this);
+    self.handleSubmit = this.handleSubmit.bind(this);
+  };
 
-        Buyer Address <input type="text" ref={ref => buyer = ref} />
-        Seller Address <input type="text" ref={ref => seller = ref} />
-        Price <input type="text" ref={ref => price = ref} />
-        <input type="submit" value="Create Agreement" />
+  handleChange(e: SyntheticInputEvent & { target: HTMLInputElement}) {
+    const { target } = e;
+    this.setState({ [target.name]: target.value });
+  }
 
-        <p>{error}</p>
+  handleSubmit(e: SyntheticInputEvent) {
+    e.preventDefault();
+    const { buyer, seller, price } = this.state;
+    if (!buyer || !seller || !price) {
+      // TODO: Add error logic/ alert dropdown
+      return alert('Form not complete');
+    };
+    this.props.onFormSubmit(buyer.trim(), seller.trim(), price.trim());
```

Comment:
The `Props` type in this file doesn't include an `onFormSubmit` function, does it?

How does this work with Flow?